docs/en/manuals/extensions_build_variants.md

```diff
@@ -0,0 +1,173 @@
+# Native Extensions - Build Variants
+
+## Build Variants
+
+When you bundle a game, you need to choose what type of engine you wish to use.
+
+  * Debug
+  * Release
+  * Headless
+
+These different versions are also referred to as `Build variants`
+
+Note: When you choose `Build and run` you'll get the debug version.
+
+### Debug
+
+This type of executable still has the debugging feature left inside it, such as:
+profiling, logging and hot reload.
+
+This variant is chosen during development of the game.
+
+### Release
+
+This variant has the debugging features disabled.
+This options is chosen when the game is ready to be released to the app store.
+
+### Headless
+
+This executable runs without any graphics. It means that you can run the game unit/smoke tests on a CI server,
+or even have it as a game server in the cloud.
+
+## App Manifest
+
+Not only can you add native code to the engine with the Native Extensions feature, you can also remove standard parts of the engine.
+E.g. if you don't need a physics engine, you can remove that from the executable.
+
+We support this via a file called an `App Manifest` (.appmanifest).
+In such a file, you can configure what libraries or symbols to remove, or perhaps add compile flags
+
+This feature is still in the works, and needs more attention.
+
+### Combined context
+
+The app manifest actually has the same structure and syntax as the extension manifest.
+This allows us to merge the contexts for one platform together when finally building.
+
+And, Defold itself, has its own build manifest as the foundation (`build.yml`).
+For each extension that is built, the manifests are combined as follows:
+
+	manifest = merge(game.appmanifest, ext.manifest, build.yml)
```

Comment:
Perhaps make it clear that this to the left overrides stuff to the right on that line?